feat: Add generate deeplink tool for dashboards, panels, and explore queries

iamrajiv opened the pull request:
Fixes: #166 

<img width="521" height="203" alt="Screenshot 2025-07-23 at 11 59 31 PM" src="https://github.com/user-attachments/assets/6ebddfae-d288-44f3-a404-67bdfca36c08" />

sd2k reviewed (commented):
Nice, @iamrajiv, this looks awesome.

I'd love to see an end-to-end test for this (in the `tests` directory), it'd be a great way to demo how it looks. Do you think you could have a go at that?

Could you also update the README with the new tool?

sd2k reviewed (changes requested):
Thanks for adding docs & tests, I'm trying to run the tests locally but they're not all passing, perhaps some of the requirements are a bit strict. Honestly they're mostly smoke tests since it's impossible to control the underlying models so you can simplify them to just include the first one or two, that should be less flaky.

iamrajiv commented:
thanks for the review @sd2k. i’ve removed the unnecessary file i had created those to test my stuff and yes we can use make run i just noticed the makefile now

also i’ve updated the readme to include this feature and we have e2e tests too which are working fine

do you think we should have a demo video as well like how to connect usage all mcp features and all that not in this pr but maybe in a separate issue and improvement?

iamrajiv commented on tools/navigation.go:
```diff
@@ -0,0 +1,102 @@
+
+type GenerateDeeplinkParams struct {
+	ResourceType string            `json:"resourceType" jsonschema:"required,description=Type of resource: dashboard\\, panel\\, or explore"`
+	UID          string            `json:"uid" jsonschema:"required,description=UID of the resource"`
```
sure @sd2k updated the field to `DashboardUID, DatasourceUID, PanelID`
